certification/internal/shell/skopeo.go

```diff
@@ -3,39 +3,71 @@ package shell
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"os/exec"
-	"strings"
+	"regexp"
 
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/errors"
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
 	log "github.com/sirupsen/logrus"
 )
 
 type SkopeoCLIEngine struct{}
 
 func (e SkopeoCLIEngine) ListTags(image string) (*cli.SkopeoListTagsReport, error) {
-	imageName := strings.Split(image, ":")[0]
-	cmd := exec.Command("skopeo", "list-tags", "docker://"+imageName)
+
+	imageName, err := e.ImageName(image)
+
+	if err != nil {
+		log.Error("unable to parse the image name", err)
+		log.Debug("error parsing the image name: ", err)
+		log.Trace("failure in attempt to parse the image name to strip tag/digest")
+		return nil, err
+	}
+
+	cmdArgs := []string{"list-tags", "docker://" + imageName}
+
+	log.Trace("running skopeo with the following invocation", cmdArgs)
+	cmd := exec.Command("skopeo", cmdArgs...)
 
 	var stdout, stderr bytes.Buffer
 	cmd.Stdout = &stdout
 	cmd.Stderr = &stderr
-	err := cmd.Run()
+	err = cmd.Run()
 
 	if err != nil {
-		return nil, err
+		return &cli.SkopeoListTagsReport{Stdout: stdout.String(), Stderr: stderr.String()}, err
 	}
 
 	var report cli.SkopeoListTagsReport
 
 	err = json.Unmarshal(stdout.Bytes(), &report)
 	if err != nil {
-		log.Error("unable to parse skopeo list-tags data for image", err)
+		log.Error("unable to parse skopeo list-tags data for image: ", err)
 		log.Debug("error marshaling skopeo list-tags data: ", err)
 		log.Trace("failure in attempt to convert the raw bytes from `skopeo list-tags` to a [map[string]interface{}")
 		return nil, err
 	}
 	report.Stdout = stdout.String()
 	report.Stderr = stderr.String()
 
+	log.Debugf(fmt.Sprintf("detected these tags for %s: %s", imageName, report.Tags))
+
 	return &report, nil
 }
+
+func (e SkopeoCLIEngine) ImageName(image string) (string, error) {
```

Comment:
Is there a reason for this to be exported?

Reply:
Ahh🤦🏻‍♀️ Fixed!